### PR TITLE
More comments

### PR DESCRIPTION
**Pull request by levinsamuel**

### CODE REVIEW

File: transparency-in-coverage/python/processors/helper_suggestions/state_management_flattener.py [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+        # Build (but don't write) the root data
+        flattener._build_root()
+
+        # Jump (fast-forward) to provider references
+        # Build (but don't write) the provider references
+        flattener.ffwd(("", "map_key", "provider_references"))
+        flattener._build_provider_references()
+        flattener._build_remote_provider_references()
+        flattener._make_provider_ref_map()
+        return flattener
```
Comment (levinsamuel) [inline transparency-in-coverage/python/processors/helper_suggestions/state_management_flattener.py:48]: So the factory idea here is to say, whatever internal moving and shifting has to be done to get this class into a usable state, encapsulate it in a method that tells the user what they are supposed to know. As far as I can tell, this is just one large process to build up this class to then be able to iterate through some data. If there are variables here, meaning different ways to do the process I have highlighted, those can then be expressed either as options to this factory method, or (perhaps clearer) as separate factory methods that tell the user something about why you'd use one over the other.

File: transparency-in-coverage/python/processors/helper_suggestions/state_management_flattener.py [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+    def ffwd(self, to_row):
+        """Jump to later in the parsing stream"""
+        while self.current_row != to_row:
+            self.current_row = next(self.parser)
+
+    def _init_parser(self, f):
+        self.parser = ijson.parse(f, use_float=True)
+
+    def _build_root(self):
```
Comment (levinsamuel) [inline transparency-in-coverage/python/processors/helper_suggestions/state_management_flattener.py:58]: All methods that therefore are not used outside this class are marked as private. This also goes a long way towards showing another user what are the "main" methods of the class that they should know about, and which are internal details they don't necessarily need to know.

File: transparency-in-coverage/python/processors/helper_suggestions/state_management_flattener.py [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+from io import FileIO
+
+import ijson
+
+from processors.helper_suggestions.hash_dict import hashdict
+from processors.helpers import MRFOpen, Row
+from processors.schema import SCHEMA
+
+
+class StateManagedFlattener:
```
Comment (levinsamuel) [inline transparency-in-coverage/python/processors/helper_suggestions/state_management_flattener.py:12]: This class is branched off yours at some previous point in time so it probably won't work if you fix yours. It's meant more as a reference to explain some ideas.

File: transparency-in-coverage/python/processors/helper_suggestions/state_management_flattener.py (hunk 2 above)
Comment (alecstein) [reply transparency-in-coverage/python/processors/helper_suggestions/state_management_flattener.py:58]: This is legit as fuck

File: transparency-in-coverage/python/processors/helper_suggestions/state_management_flattener.py (hunk 1 above)
Comment (alecstein) [reply transparency-in-coverage/python/processors/helper_suggestions/state_management_flattener.py:48]: I went ahead and did this in a slightly different way, gonna text you

**Pull request closed without merging**